File: src/main/java/br/com/controle/financeiro/model/dto/ClientDTO.java
package br.com.controle.financeiro.model.dto;

import java.io.Serializable;
import java.util.UUID;

import javax.validation.constraints.NotBlank;

import br.com.controle.financeiro.model.entity.Client;
import br.com.controle.financeiro.model.entity.UserEntity;
import com.fasterxml.jackson.annotation.JsonProperty;

public class ClientDTO implements Serializable {

    @NotBlank
    private String name;

    private UUID client;

    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private String owner;

    public ClientDTO() {
        super();
    }

    private ClientDTO(final UUID id, final String name, final String owner) {
        super();
        this.client = id;
        this.name = name;
        this.owner = owner;
    }

    public static ClientDTO fromClient(Client client) {
        return new ClientDTO(client.getId(), client.getName(), client.getOwner().getUsername());
    }

    public Client toClient(UserEntity owner) {
        return new Client(this.client, this.name, owner);
    }

    public UUID getId() {
        return this.client;
    }

    public void setId(final UUID id) {
        this.client = id;
    }

    public String getName() {
        return this.name;
    }

    public void setName(final String name) {
        this.name = name;
    }

    public String getOwner() {
        return owner;
    }

    public void setOwner(String owner) {
        this.owner = owner;
    }

}

File: src/main/kotlin/br/com/controle/financeiro/model/dto/ClientDTO.kt
package br.com.controle.financeiro.model.dto

import br.com.controle.financeiro.model.entity.Client
import br.com.controle.financeiro.model.entity.UserEntity
import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.constraints.NotBlank
import java.io.Serializable
import java.util.UUID

data class ClientDTO(
        @NotBlank val name: String,
        @JsonProperty(access = JsonProperty.Access.READ_ONLY) val owner: UserEntity,
        val id: UUID? = null,
) : Serializable {

    fun toClient(owner: UserEntity): Client {
        return Client(name, owner, id)
    }

    companion object {
        fun fromClient(client: Client): ClientDTO {
            return ClientDTO(client.name, client.owner, client.id)
        }
    }
}
